plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,268 @@
+		optionalAccountIDs:           optionalAccountIDs,
+	}
+}
+
+func makeWantVolumeFilter(config *VolumeFilterConfig, firstMarketID string, marketIDs []string, optionalAccountIDs []string, action string) *volumeFilter {
+	queryMarketIDs := utils.Dedupe(append([]string{firstMarketID}, marketIDs...))
+	query, e := queries.MakeDailyVolumeByDateForMarketIdsAction(&sql.DB{}, queryMarketIDs, action, optionalAccountIDs)
+	if e != nil {
+		panic(e)
+	}
+
+	return &volumeFilter{
+		name:                   "volumeFilter",
+		baseAsset:              utils.NativeAsset,
+		quoteAsset:             utils.NativeAsset,
+		config:                 config,
+		dailyVolumeByDateQuery: query,
+	}
+}
+
+func TestMakeFilterVolume(t *testing.T) {
+	testAssetDisplayFn := model.MakeSdexMappedAssetDisplayFn(map[model.Asset]hProtocol.Asset{model.Asset("XLM"): utils.NativeAsset})
+	configValue := ""
+	exchangeName := ""
+	tradingPair := &model.TradingPair{Base: "XLM", Quote: "XLM"}
+	modes := []volumeFilterMode{volumeFilterModeExact, volumeFilterModeIgnore}
+	firstMarketID := MakeMarketID(exchangeName, "native", "native")
+
+	testCases := []struct {
+		name       string
+		marketIDs  []string
+		accountIDs []string
+		wantFilter *volumeFilter
+	}{
+		// TODO DS Confirm the empty config fails once validation is added to the constructor
+		{
+			name:       "1 market id",
+			marketIDs:  []string{"marketID"},
+			accountIDs: []string{},
+		},
+		{
+			name:       "2 market ids",
+			marketIDs:  []string{"marketID1", "marketID2"},
+			accountIDs: []string{},
+		},
+		{
+			name:       "2 dupe market ids, 1 distinct",
+			marketIDs:  []string{"marketID1", "marketID1", "marketID2"},
+			accountIDs: []string{},
+		},
+		{
+			name:       "1 account id",
+			marketIDs:  []string{},
+			accountIDs: []string{"accountID"},
+		},
+		{
+			name:       "2 account ids",
+			marketIDs:  []string{},
+			accountIDs: []string{"accountID1", "accountID2"},
+		},
+		{
+			name:       "account and market ids",
+			marketIDs:  []string{"marketID"},
+			accountIDs: []string{"accountID"},
+		},
+	}
+
+	for _, k := range testCases {
+		// this lets us test both types of modes when varying the market and account ids
+		for _, m := range modes {
+			// this lets us test both constraints within the config
+			baseCapInBaseConfig := makeTestVolumeFilterConfig(1.0, -1.0, k.marketIDs, k.accountIDs, m)
+			baseCapInQuoteConfig := makeTestVolumeFilterConfig(-1.0, 1.0, k.marketIDs, k.accountIDs, m)
+
+			for _, config := range []*VolumeFilterConfig{baseCapInBaseConfig, baseCapInQuoteConfig} {
+				// configType is used to represent the type of config when printing test name
+				var configType string
+				if config.SellBaseAssetCapInBaseUnits != nil {
+					configType = "base"
+				} else {
+					configType = "quote"
+				}
+
+				// TODO DS Vary filter action between buy and sell, once buy logic is implemented.
+				wantFilter := makeWantVolumeFilter(config, firstMarketID, k.marketIDs, k.accountIDs, "sell")
```

Comment:
I get the feeling we had discussed this in the last review -- this cannot use the same `marketIDs` and `accountIDs` variable from the input to construct the `want` value.

simple example:
input marketIDs = []string{"marketID1", "marketID1"}
wantMarketIDs = []string{"marketID1"}

therefore these variables cannot be the same.

Reply:
This doesn't do that, because the `firstMarketID` is pre-pended to `k.marketIDs` for the comparison.

The reason `wantFilter` cannot go on the test cases is here: https://github.com/stellar/kelp/pull/552#discussion_r514447104

There's no way for us to construct the `config`, which we loop over, if the `marketIDs` and `accountIDs` aren't already defined. But those are defined within the test case. And we can't construct the `want` value without the `config`. Thoughts on how to solve this?

Reply:
noting here, let's discuss on a call early next week on Monday or Tuesday and merge this PR in a pair programming session